pkg/logger/logger.go

```diff
@@ -35,6 +35,35 @@ func SetGlobalLogLevel(level int) {
 	logLevel = level
 }
 
+func SetLogLevelByName(level string) (err error, msg string) {
+	newLevel, err := GetLogLevelByName(level)
+	msg = "Log level not changed. Staying "
+	if err != nil {
+		return err, msg
+	}
+
+	existingLevel := GetLogLevel()
+	if newLevel != existingLevel {
```

Comment:
In order to avoid indentation and a few cpu cycles, check the opposite condition first and return faster